test/test_torch.py

```diff
@@ -18021,6 +18020,48 @@ def is_view_of(self, base, other):
 
         return True
 
+    def transpose_if_not_contig(self, x, contiguous=False, dim0=0, dim1=1):
```

Comment:
_transpose_if_not_contig (helper functions typically start their names with an underscore)

Reply:
Also don't take defaults. If you want to require contiguous, dim, etc. be specified with keywords just put them after a *. Defaults are appropriate for user-facing functions, but I think this is developer facing?

Reply:
This function has a misleading name that suggests it's checking the contiguity of x. Maybe get rid of it and just enumerate `do_transpose in (True, False):` in the tests below?

Reply:
updated